pulsar-client/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -240,6 +244,60 @@ public ProducerConfiguration setBatchingEnabled(boolean batchMessagesEnabled) {
     }
 
     /**
+     * @return the CryptoKeyReader
+     */
+    public CryptoKeyReader getCryptoKeyReader() {
+        return this.cryptoKeyReader;
+    }
+
+    /**
+     * Sets a {@link CryptoKeyReader}
+     *
+     * @param cryptoKeyReader
+     *            CryptoKeyReader object
+     */
+    public ProducerConfiguration setCryptoKeyReader(CryptoKeyReader cryptoKeyReader) {
+        checkNotNull(cryptoKeyReader);
+        this.cryptoKeyReader = cryptoKeyReader;
+        return this;
+    }
+
+    /**
+     * 
+     * @return encryptionKeys
+     *  
+     */
+    public  ConcurrentOpenHashSet<String> getEncryptionKeys() {
+        return this.encryptionKeys;
+    }
+
+    /**
+     * 
+     * Returns true if encryption keys are added
+     *  
+     */
+    public boolean isEncryptionEnabled() {
+        return (this.encryptionKeys != null) && !this.encryptionKeys.isEmpty();
+    }
+
+    /**
+     * Add public encryption key, used by producer to encrypt the data key.
+     *
+     * At the time of producer creation, Pulsar client checks if there are keys added to encryptionKeys.
+     * If keys are found, a callback getKey(String keyName) is invoked against each key to load
+     * the values of the key. Application should implement this callback to return the key in pkcs8 format.
+     * If compression is enabled, message is encrypted after compression.
+     * If batch messaging is enabled, the batched message is encrypted.
+     *
+     */
+    public void addEncryptionKey(String key) {
```

Comment:
Need to mention that this option is not to be used if batch messaging is enables and add appropriate asserts.

Reply:
Batch message will work as long as proper keys are used. It's a problem only for cases with decryption failure, which is not very different from non batch message failure.